### PR TITLE
Renamed ManagedZoneInfo to ZoneInfo and added time units for ttl. 

### LINKED ISSUE #579: Rename ManagedZone to Zone
As per suggestion by @jgeewax , we should rename ManagedZone to just Zone. The same thing holds for the metadata object ManageZoneInfo to ZoneInfo. This concern documentation too.

### LINKED ISSUE #581: Change TTL to accept unit
As per suggestion of @jgeewax, change ttl in DnsRecord to accept duration as a number and TimeUnit. For an example, see e.g. here:

https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java#L1475

### PR DESCRIPTION
**Pull request by mderka**
Fixes #581 and fixes #579.

### CODE REVIEW

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java [hunk 1]
```diff
@@ -176,14 +177,23 @@ public Builder name(String name) {
...
-    public Builder ttl(int ttl) {
-      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was %s.", ttl);
-      this.ttl = ttl;
+    public Builder ttl(int duration, TimeUnit unit) {
+      checkArgument(duration >= 0,
+          "Duration cannot be negative. The supplied value was %s.", duration);
+      checkNotNull(unit);
+      // convert to seconds and check that we are not overflowing int
+      // we cannot do that because pb does not support it
+      long converted = unit.toSeconds(duration);
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java]: Could be done by `Ints.checkedCast`(http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/primitives/Ints.html#checkedCast(long))

**Comment by mderka** [comment]
Ready for another pass.

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ZoneInfo.java [hunk 2]
```diff
@@ -174,63 +172,63 @@ private ManagedZoneInfo(Builder builder) {
...
-   * Returns the read-only managed zone id assigned by the server.
+   * Returns the read-only zone id assigned by the server.
    */
   public BigInteger id() {
     return id;
   }
 
   /**
-   * Returns the time when this time that this managed zone was created on the server.
+   * Returns the time when this time that this zone was created on the server.
```
Comment (aozarov) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ZoneInfo.java]: "Returns the time when this zone was created." ?

**Comment by aozarov** [comment]
Looks good. one comment (also just noticed that in `ProjectInfoTest` we refer to the static `fromPb` method using an instance).

File: gcloud-java-dns/src/main/java/com/google/gcloud/dns/ZoneInfo.java (hunk 2 above)
Comment (mderka) [inline gcloud-java-dns/src/main/java/com/google/gcloud/dns/ZoneInfo.java]: Fixed.

**Comment by mderka** [comment]
Fixed ProjectInfoTest too.

**Pull request merged**